kernel/msg_q.c

```diff
@@ -195,6 +197,83 @@ static inline int z_vrfy_k_msgq_put(struct k_msgq *msgq, const void *data,
 #include <syscalls/k_msgq_put_mrsh.c>
 #endif /* CONFIG_USERSPACE */
 
+int z_impl_k_msgq_put_front(struct k_msgq *msgq, const void *data, k_timeout_t timeout)
+{
+	__ASSERT(!arch_is_in_isr() || K_TIMEOUT_EQ(timeout, K_NO_WAIT), "");
+
+	struct k_thread *pending_thread;
+	k_spinlock_key_t key;
+	int result;
+
+	key = k_spin_lock(&msgq->lock);
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_msgq, put, msgq, timeout);
```

Comment:
It might be a good idea to update the parameters to SYS_PORT_TRACING_OBJ_FUNC_xxx() used in this routine.

Reply:
Done.